docs/plugin/subsystems.md

```diff
@@ -0,0 +1,288 @@
+Thunder provides "Subsystems" which are abstract categories of functionality (Network, Graphics, Internet) that can be marked as active/inactive by plugins. This mechanism allows plugins to delay activation until certain conditions are met, or ensure that plugins are deactivated in a suitable order.
+
+For example, a web-browser plugin could be prevented from starting until the Internet and Graphics subsystems are ready. The Graphics subsystem could be marked as active by a network management plugin, and the graphics by a window manager/compositor plugin.
```

Comment:
"The Graphics subsystem could be marked as active by a network management plugin" This looks like a copy-paste msitake ;-) I guess you mean: "The Network (or Internet) subsystem could be marked as active by a network management plugin"
Here (at metrological embedded development) we assume that management plugins are *useless* plugins and contradict the philosophy of OO where a class/plugins implements/realizes functionality and does not relay it to other classes/plugins. 
Managers, as known in the context of work, are entities that act as mail routers. They do not realize functionality, they just distribute work and this should not be used in (embedded) software. It is overhead that consumes the limited resources these systems have. In embedded software we are only looking for entities that *realize* functionality, not move it on to the next plugin. 
There are *no* Manager plugins found in ThunderNanoServices as each plugin realizes some functionality. Look at the amount of Managers you find in RDKServices :-D